pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -42,21 +42,47 @@ var (
 )
 
 type tinkerbellProvider struct {
-	clusterConfig    *v1alpha1.Cluster
-	datacenterConfig *v1alpha1.TinkerbellDatacenterConfig
+	clusterConfig          *v1alpha1.Cluster
+	datacenterConfig       *v1alpha1.TinkerbellDatacenterConfig
+	machineConfigs         map[string]*v1alpha1.TinkerbellMachineConfig
+	controlPlaneSshAuthKey string
+	workerSshAuthKey       string
+	// etcdSshAuthKey         string
+	providerKubectlClient ProviderKubectlClient
+	templateBuilder       *TinkerbellTemplateBuilder
+	hardwareConfig        string
```

Comment:
This would be the path to a yaml file that contains the Hardware CRDs, right? Clearer to call it hardwareConfigFile?